welcome_wizard/views.py

```diff
@@ -48,140 +58,182 @@ def check_sync(instance, request):
+            for obj in self.queryset.filter(pk__in=pk_list):
+                JobResult.enqueue_job(job_model=job, user=request.user, manufacturer_name=obj.name)
+                onboarded += 1
+            messages.success(request, f"Onboarded {onboarded} objects.")
+        return redirect("plugins:welcome_wizard:manufacturerimport_list")
+
+
+class DeviceTypeImportUIViewSet(
+    ObjectDetailViewMixin,
+    ObjectListViewMixin,
+    ObjectChangeLogViewMixin,
+):  # pylint: disable=abstract-method
+    """List view for DeviceTypeImport."""
 
     permission_required = "welcome_wizard.view_devicetypeimport"
-    table = DeviceTypeImportTable
-    queryset = DeviceTypeImport.objects.prefetch_related("manufacturer")
-    filterset = DeviceTypeImportFilterSet
+    queryset = DeviceTypeImport.objects.select_related("manufacturer")
+    table_class = DeviceTypeImportTable
+    filterset_class = DeviceTypeImportFilterSet
+    filterset_form_class = DeviceTypeImportFilterForm
     action_buttons = ()
-    template_name = "welcome_wizard/devicetype.html"
-    filterset_form = DeviceTypeImportFilterForm
-
-    def get(self, request, *args, **kwargs):
-        """Add Check Sync to Get."""
-        check_sync(instance=self, request=request)
-        return super().get(request, *args, **kwargs)
-
-
-class BulkImportView(View, ObjectPermissionRequiredMixin):
-    """Generic Bulk Import View."""
-
-    return_url = None
-    model = None
-    form = forms.Form
-    bulk_import_url = None
-    breadcrumb_name = None
-    _permission_action = None
-
-    def get_required_permission(self):
-        """Return the specific permission necessary to perform the requested action on an object."""
-        return get_permission_for_model(self.queryset.model, self._permission_action)
-
-    def dispatch(self, request, *args, **kwargs):
-        """Ensures User has permission to add."""
-        # Following Nautobot style of adding self._permission_action inside the dispatch.
-        self._permission_action = "add"  # pylint disable=attribute-defined-outside-init
-        return super().dispatch(request, *args, **kwargs)
-
-    def get(self, request):
-        """Single Import Page."""
-        # The user is expected to get here only by selecting the Import button from a list view.
-        if not request.GET.get("pk"):
-            return redirect(self.return_url)
-        primarykey = request.GET["pk"]
-        initial = {"pk": [primarykey]}
-
-        obj = self.model.objects.get(pk=primarykey)
-
-        form = self.form(initial)
-
-        return render(
-            request,
-            "welcome_wizard/import.html",
-            {
-                "form": form,
-                "obj": obj,
-                "return_url": self.return_url,
-                "bulk_import_url": self.bulk_import_url,
-                "breadcrumb_name": self.breadcrumb_name,
-            },
-        )
+    serializer_class = serializers.Serializer
```

Comment:
This should be using `welcome_wizard.api.serializers.DeviceTypeImportSerializer`